backend/de.metas.async/src/main/java/de/metas/lock/spi/impl/SqlLockDatabase.java

```diff
@@ -245,11 +263,92 @@ protected int performLockSQLInsert(final ILockCommand lockCommand, final List<Ob
 		}
 	}
 
-	@Override
-	protected boolean lockRecord(final ILockCommand lockCommand, final TableRecordReference record)
+	private ImmutableList<ExistingLockInfo> retrieveExistingLocksForFilter(final int adTableId, @NonNull final ISqlQueryFilter filter)
 	{
-		Check.assumeNotNull(record, "record not null");
+		return retrieveExistingLocksForWhereClause(
+				adTableId,
+				filter.getSql(),
+				filter.getSqlParams(null));
+	}
 
+	private ImmutableList<ExistingLockInfo> retrieveExistingLocksForSelection(final int adTableId, @NonNull final PInstanceId pinstanceId)
+	{
+		return retrieveExistingLocksForWhereClause(
+				adTableId,
+				I_T_Lock.COLUMNNAME_Record_ID + " IN (select T_Selection_ID from T_Selection WHERE AD_PInstance_ID=" + pinstanceId.getRepoId() + ")",
+				null);
+	}
+
+	private ImmutableList<ExistingLockInfo> retrieveExistingLocksForWhereClause(
+			final int adTableId,
+			@NonNull final String whereSql,
+			@Nullable final List<Object> whereSqlParams)
+	{
+		final String sql =
+				"SELECT " + I_T_Lock.COLUMNNAME_AD_Table_ID + ", "
+						+ I_T_Lock.COLUMNNAME_T_Lock_ID + ", "
+						+ I_T_Lock.COLUMNNAME_Record_ID + ", "
+						+ I_T_Lock.COLUMNNAME_Owner + ", "
+						+ I_T_Lock.COLUMNNAME_IsAutoCleanup + ", "
+						+ I_T_Lock.COLUMNNAME_IsAllowMultipleOwners + ", "
+						+ " FROM " + I_T_Lock.Table_Name
+						+ " WHERE " + I_T_Lock.COLUMNNAME_AD_Table_ID + " = " + adTableId
+						+ " AND " + whereSql;
+		PreparedStatement pstmt = null;
+		ResultSet rs = null;
+		try
+		{
+			pstmt = DB.prepareStatement(sql, ITrx.TRXNAME_None);
```

Comment:
use org.compiere.util.DB#retrieveRows(java.lang.CharSequence, java.util.List<java.lang.Object>, org.compiere.util.DB.ResultSetRowLoader<T>)

Reply:
aweome, thx!

Reply:
@metas-ts  can u pls also fix this, just saw it 🤦‍♂️ 
![image](https://user-images.githubusercontent.com/1244701/120594472-a849d600-c449-11eb-909a-7aec2744ac61.png)

Reply:
done